### PR TITLE
feat: Manage Collectibles panel UI component & model

### PR DESCRIPTION
**Pull request by caybro**
- implements a QML panel to organize collectibles
- factors out the delegates to separate files to be reusable with the Assets tab
- adds QML tests to assess the UI functionality (move, show/hide, save/load)
- does NOT cover the problematic "Arrange by collection" switch (will be handled separately in https://github.com/status-im/status-desktop/issues/12703)

Fixes: https://github.com/status-im/status-desktop/issues/12379

### What does the PR do

Adds a Manage Collectibles UI panel component to storybook

### Affected areas

storybook

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

image

### CODE REVIEW

File: storybook/qmlTests/main.cpp
Comment (caybro) [inline storybook/qmlTests/main.cpp:26]: Wee don't want to overwrite user's actual settings :)

**Review by alexandraB99 — APPROVED** [review]
massive! is the commented code in ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml still there on purpose?

**Comment by caybro** [comment]
> massive! is the commented code in ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml still there on purpose?

Yup, I'll create a separate issue to handle/enable that later

File: storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml
Comment (micieslak) [inline storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml:187]: I wonder if maybe instead of using fixed value like `80` it could be possible to use value derived from delegate height, to make the test more resistant to pure-ui changes like changing delegate's height.

File: storybook/src/Models/ManageCollectiblesModel.qml
Comment (micieslak) [inline storybook/src/Models/ManageCollectiblesModel.qml:20]: 1. Referring to some external resources is a bit risky, the page may become unavailable, the content may change and copyrights may be violated. So ideally we should refer to some owned resources.
2. We keep some longer resources in ModelData, maybe putting some images here as base64 would be a solution.

**Review by micieslak — COMMENTED** [review]
LGTM in general, some minor comments mainly. Nice to have some qml tests here :). Regarding models @noeliaSD may want to take a look.

File: ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml:73]: This is https://github.com/status-im/status-desktop/issues/12703, will add a TODO/comment

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:44]: Yeah, I can drop that (pun intended :D

File: storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml
Comment (caybro) [reply storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml:187]: Yeah, I was also thinking about that... but it's hard, the height can be variable

File: storybook/src/Models/ManageCollectiblesModel.qml
Comment (caybro) [reply storybook/src/Models/ManageCollectiblesModel.qml:20]: Hmmm... it's definitely not a (c) violation, it would be if we included it which I didn't want (in order not to bloat the already big sources). True that it might disappear any time bug really not a big deal imo

File: storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml
Comment (caybro) [reply storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml:187]: But it's fine imo... if we ever change the delegate height for whatever reason, this test will start failing so we definitely notice

File: ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml:74]: Fixed

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: Fixed

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: Fixed

File: ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml]: Fixed

File: ui/StatusQ/src/wallet/managetokenscontroller.cpp
Comment (micieslak) [reply ui/StatusQ/src/wallet/managetokenscontroller.cpp:250]: Yeah, I know, only aesthetic aspect, `!!!` looks like some accidentally left debug print.

File: storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml
Comment (micieslak) [reply storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml:187]: > But it's fine imo... if we ever change the delegate height for whatever reason, this test will start failing so we definitely notice

Yea, we will notice but will need to fix. But if it's problematic, than ok, nothing important.

File: storybook/src/Models/ManageCollectiblesModel.qml
Comment (micieslak) [reply storybook/src/Models/ManageCollectiblesModel.qml:20]: As you prefer. I'm just pointing out that there is already a certain convention for this type of thing.

**Review by micieslak — APPROVED** [review]
LGTM, but please ask @noeliaSD if she wants to review as well before merging :)

**Review by noeliaSD — COMMENTED** [review]
Hey, I've done a visual / testing review first and meanwhile, I leave some comments / questions.. Following with the code review!

 1. I found some issues with the reordering of the group community and hidding / showing individual community collectibles. 
There isn't a pattern I can define but here there is 1 video where the issue happens. The origin can be, from hide to show or the other way around. 

- **Second 8 to 12** --> Group Pandas is in the last position, there's a collectible from this group that we hide, then the group is moved to the top position.

- **Second 20 to 25** --> Group Bearz is on top, there's a collectible from this group that is hidden and we show it, then the group is moved to the last position.  

https://github.com/status-im/status-desktop/assets/97019400/13cb9207-f9c0-4812-a290-5e6ef02cb51d

2. Question: Looking at design there's this requirement:

`Arranging by community groups all collectibles into their respective communities:
- Collections with most collectibles are placed first 
- Collectibles retain their respective order from previous view (we can’t assume the user wants to change respective ordering)`

Following this requirement, I understand that we should always place on top of the groups list, the one that has more collectibles each time the `arrenge by community` is switched on. Here, I don't see it's done like this, but by using the first collectible found in the flat collectibles list order. See video for better understanding:

https://github.com/status-im/status-desktop/assets/97019400/df9b7006-1b57-481a-b1b9-6f0b21e990c1

**Comment by caybro** [comment]
> Hey, I've done a visual / testing review first and meanwhile, I leave some comments / questions.. Following with the code review!
> 
> 1. I found some issues with the reordering of the group community and hidding / showing individual community collectibles.
>    There isn't a pattern I can define but here there is 1 video where the issue happens. The origin can be, from hide to show or the other way around.
> 
> * **Second 8 to 12** --> Group Pandas is in the last position, there's a collectible from this group that we hide, then the group is moved to the top position.
> * **Second 20 to 25** --> Group Bearz is on top, there's a collectible from this group that is hidden and we show it, then the group is moved to the last position.

Yeah, I can reproduce, will fix and add a test, thanks! :)

> `Arranging by community groups all collectibles into their respective communities:
> 
> * Collections with most collectibles are placed first
> * Collectibles retain their respective order from previous view (we can’t assume the user wants to change respective ordering)`
> 
> Following this requirement, I understand that we should always place on top of the groups list, the one that has more collectibles each time the `arrenge by community` is switched on. Here, I don't see it's done like this, but by using the first collectible found in the flat collectibles list order. See video for better understanding:

So the requirement is that this applies only the first time user enters the screen (before any changes are made); after that, it's the custom order that the user sets. And yeah, it's true that currently we don't (pre)sort the community collectibles that way, can you pls file a separate issue? I'll handle it in the followup PR

File: ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml
Comment (noeliaSD) [inline ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:24]: Could you explain what do you refer in this comment? I'm not sure if I'm understanding what the `childCount` is expected to represent.

File: ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml
Comment (noeliaSD) [inline ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:57]: Leftover?

File: ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml
Comment (noeliaSD) [inline ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml:49]: I'm not sure about that.. Isn't the base model going to have a role called  `symbol `yet? Is / should this `uid` be the `key` role to uniqually identify the item?

File: ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml
Comment (noeliaSD) [inline ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml:168]: Leftover?

**Review by noeliaSD — COMMENTED** [review]
Code review done! 

Overall LGTM, just left some comments / questions!

**Comment by noeliaSD** [comment]
> So the requirement is that this applies only the first time user enters the screen (before any changes are made); after that, it's the custom order that the user sets. And yeah, it's true that currently we don't (pre)sort the community collectibles that way, can you pls file a separate issue? I'll handle it in the followup PR

Here the task: https://github.com/status-im/status-desktop/issues/12766

File: ui/StatusQ/src/wallet/managetokensmodel.h
Comment (caybro) [reply ui/StatusQ/src/wallet/managetokensmodel.h:30]: I can add it in the next integration PR (already WIP), this one is really just UI (QML) work mostly, didn't want to introduce any model changes

File: ui/app/AppLayouts/Wallet/controls/ManageTokenMenuButton.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokenMenuButton.qml:19]: Yup will do, next PR :)

File: ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml:1]: It already is, via both ManageTokensPanel.qml and ManageCollectiblesPanel.qml, doesn't make much sense to have it standalone imo

File: ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml:32]: Will do but we have nothing for `12` or `4` :/

File: ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml:14]: I can't see the advantage? Imo `property string text` is more readable (gives you the type of property right in the header of the file), you can have default values and don't have to introduce `id`s to other elements

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:59]: Yup, my visual debugging helper :)

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:66]: I get the point but I don't have access to `Constants.tokenIcon(model.symbol)` in C++; I'm hoping one of the new models could provide that for both assets and collectibles; will add a TODO here to unify

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:64]: Pls a separate issue :) Mostly because we don't have the information from the backend atm, right?

File: ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:24]: The number of child delegates (ie the tokens belonging to this group); will add a TODO to the model to add an extra role

File: ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml:32]: Done

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:1]: Done

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: Removed

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: Done

File: ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:1]: Done

File: ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml]: Removed

File: ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml]: Removed

File: storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml
Comment (caybro) [inline storybook/qmlTests/tests/tst_ManageCollectiblesPanel.qml:289]: And here's the 60 line test for that bug :)

File: ui/StatusQ/src/wallet/managetokensmodel.h
Comment (noeliaSD) [reply ui/StatusQ/src/wallet/managetokensmodel.h:30]: Ok! It's just a reminder to introduce it earlier since I'd say that after, it will be more complicated!! But yeah, it's perfect if you do it in the next iteration!

File: ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:26]: Ok, but not the total balance? just the number of delegates inside the group, right?

File: ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/panels/ManageCollectiblesPanel.qml:49]: Ok so.. `uid` contains information about the `symbol`, so we should also keep the `key` and the `symbol`. The `key`(although in the major cases will be == to`symbol` value) will uniqualy identify the element as said before!

File: ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml:1]: Up to you! May be used in some point in other context!

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: Yes, assume the model you will provide as the source will provide already this info!

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: Well, to visualize it in the collectibles view, YES, we already have it, so it's question of checking which is the model used in the main collectibles view and if it's the same than is used in here (it should be the same model, or at least we should try to use the same source model just bc of having better unification and simplicity), the info is alredy provided!

Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: Review `CollectiblesView.qml` and `CollectibleView.qml` files. I think that the interesting roles in here are: `communityPrivilegesLevel` and `communityColor` roles.

Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: You can perfectly do it in a spearate task if you prefer!

File: ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:26]: Yup, correct

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: "Will" but not yet :) Once it does, I'll add it to my controller/model

Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: It's a matter of switching or transforming the baseModel, in which case I'll also adjust the role names; remember we have tests that will start failing immediately if the base model changes w/o us adjusting the C++ controller/model

**Review by noeliaSD — APPROVED** [review]
LGTM!! Thanks for addressing the comments!

Tested again and the issue has been solved!!

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: Ok. Could you create a separate task with the pending stuff for this topic??

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:15]: Thanks for the review! Added the TODO, will be in the next PR :)

File: ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml]: @caybro has this task been created?

**Pull request merged**
